src/System.Reflection/tests/FieldInfo/FieldInfo_ConstantTests.cs

```diff
@@ -11,67 +11,36 @@ namespace System.Reflection.Tests
 {
     public class FieldInfoConstantTests
     {
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constIntField()
-        {
-            TestSetValue_constantField("constIntField", (object)222);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constStrField()
-        {
-            TestSetValue_constantField("constStrField", (object)"new value");
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constCharField()
+        public static IEnumerable<object[]> FieldInfoConstants_TestData()
         {
-            TestSetValue_constantField("charField", (object)'A');
+            yield return new object[] { "constIntField", 222 };
+            yield return new object[] { "constStrField", "new value" };
+            yield return new object[] { "charField", 'A' };
+            yield return new object[] { "boolField", false };
+            yield return new object[] { "floatField", 4.56 };
+            yield return new object[] { "doubleField", double.MaxValue };
+            yield return new object[] { "int64Field", long.MaxValue };
+            yield return new object[] { "byteField", byte.MaxValue };
         }
 
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constboolField()
-        {
-            TestSetValue_constantField("boolField", (object)false);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constfloatField()
-        {
-            TestSetValue_constantField("floatField", (object)4.56);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constdoubleField()
+        // Verify SetValue method cannot set value for constant Fields using Reflection
```

Comment:
Do we need these comments? Perhaps we could incorporate them into the test name:

E.g this method could be renamed it to `SetValue_ConstantField_ThrowsFieldAccessException`
We probably don't need the prefix Test either but that's not too important